src/components/QaccRoundEndBanner.tsx

```diff
@@ -1,9 +1,25 @@
-import React from 'react';
-import { isAllocationDone } from '@/config/configuration';
+import React, { useEffect, useState } from 'react';
+import { useFetchAllProjects } from '@/hooks/useFetchAllProjects';
+import { useFetchAllRound } from '@/hooks/useFetchAllRound';
 import QaccRoundStats from './QaccRoundStats';
+import { useCheckProjectPriceStatus } from '@/hooks/useCheckProjectPriceStatus ';
 
 interface QaccRoundEndBannerProps {}
 const QaccRoundEndBanner: React.FC<QaccRoundEndBannerProps> = ({}) => {
+  const { data: allProjects } = useFetchAllProjects();
+  const { data: allRounds } = useFetchAllRound();
+  const [isAllocationDone, setIsAllocationDone] = useState(false);
+
+  const { data: projectPriceUpdated } = useCheckProjectPriceStatus(
+    allProjects,
+    allRounds,
+  );
+
+  useEffect(() => {
+    console.log('IsAllocation Done', projectPriceUpdated);
+    setIsAllocationDone(!!projectPriceUpdated);
+  }, [projectPriceUpdated]);
```

Comment:
I think that useEffect doesn't improve efficiency. As I know, the component will rerender already due to `projectPriceUpdated` update.
```suggestion
  const { data: projectPriceUpdated } = useCheckProjectPriceStatus(
    allProjects,
    allRounds,
  );

  const isAllocationDone = !!projectPriceUpdated;
```

Reply:
<img width="1152" alt="image" src="https://github.com/user-attachments/assets/80805905-9aaa-42d6-8ece-78cdaca0cac6" />

Reply:
ok thank you @aminlatifi ...really helpful